modules/runtime/nuxeo-runtime-migration/src/main/java/org/nuxeo/runtime/migration/MigrationStep.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nour AL KOTOB
+ */
+package org.nuxeo.runtime.migration;
+
+import org.nuxeo.runtime.migration.MigrationDescriptor.MigrationStepDescriptor;
+
+/**
+ * @since 11.1
+ */
+public class MigrationStep {
+
+    protected final String id;
+
+    protected final String description;
+
+    protected final String descriptionLabel;
+
+    protected final String fromState;
+
+    protected final String toState;
+
+    public final String getId() {
+        return id;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public String getDescriptionLabel() {
+        return descriptionLabel;
+    }
+
+    public String getFromState() {
+        return fromState;
+    }
+
+    public String getToState() {
+        return toState;
+    }
+
+    public MigrationStep(MigrationStepDescriptor descriptor) {
+        this.id = descriptor.getId();
+        this.description = descriptor.getDescription();
+        this.descriptionLabel = descriptor.getDescriptionLabel();
+        this.fromState = descriptor.getFromState();
+        this.toState = descriptor.getToState();
+    }
```

Comment:
Maybe worth doing the same thing as for `Migration`, a static `from` and only simple params in the constructor.

Reply:
here you go!